### PR TITLE
Modifies exp() and pow() and specs behaviour of z^inf and inf^z.

### PR DESCRIPTION
**Pull request by harrysarson**
Previous behaviour was inconsistant and was not tested for. This
commit ensures that z^inf and inf^z return reasonable and consistant
values.

See #24 for the dicussion on the behaviour which can be summaried as:

    z ^ Infinity === NaN
    Infinity ^ z === Infinity if Im(z) === 0 and Re(z) > 0
    Infinity ^ z === 0 if Re(z) < 0
    Infinity ^ 0 === 1
    Infinity ^ z === NaN otherwise

### CODE REVIEW

File: complex.js [hunk 1]
```diff
@@ -456,6 +456,18 @@
...
       }
 
+      if (!z.isFinite()) {
+        return Complex['NAN'];
+      }
+
+      if (!this.isFinite()) {
+        return z['re'] < 0
+          ? Complex['ZERO']
+          : z['im'] === 0 && z['re'] > 0 // should we epsilon ball here
```
Comment (harrysarson) [inline complex.js]: Should the imaginary part of z be compared exactly to `0` here or should `Complex['EPISLON']` be used?

**Comment by harrysarson** [comment]
@infusion could you take a look?

**Comment by harrysarson** [comment]
@infusion I have solved the merge conflicts, would it be possible to review this?

**Comment by harrysarson** [comment]
@infusion any updates?

**Comment by infusion** [comment]
Sorry, github autoclosed this issue

**Pull request closed without merging**